app/src/androidTest/java/com/github/houseorganizer/houseorganizer/GroceriesActivityTest.java

```diff
@@ -0,0 +1,136 @@
+
+    @BeforeClass
+    public static void createFirebase() throws ExecutionException, InterruptedException {
+        FirebaseTestsHelper.startAuthEmulator();
+        FirebaseTestsHelper.startFirestoreEmulator();
+        FirebaseTestsHelper.setUpFirebase();
+        auth = FirebaseAuth.getInstance();
+    }
+
+    @AfterClass
+    public static void signOut(){auth.signOut();}
+
+    @Before
+    public void openActivity() throws InterruptedException {
+        onView(withId(R.id.house_imageButton)).perform(click());
+        onView(withId(R.id.housesView))
+                .perform(RecyclerViewActions.actionOnItemAtPosition(0, click()));
+        onView(withId(R.id.nav_bar_cart)).perform(click());
+        Thread.sleep(300);
+    }
+
+    private void addNewItem(String name, int quantity, String unit){
+        onView(withId(R.id.groceries_add)).perform(click());
+        onView(withId(R.id.edit_text_name)).perform(typeText(name));
+        onView(withId(R.id.edit_text_quantity)).perform(typeText(""+quantity));
+        onView(withId(R.id.edit_text_unit)).perform(typeText(unit));
+        onView(withText(R.string.add)).perform(click());
+    }
+
+    @Test
+    public void addTaskButtonAvailable(){
+        onView(withId(R.id.groceries_add)).check(matches(isEnabled()));
+        onView(withId(R.id.groceries_add)).check(matches(isDisplayed()));
+        onView(withId(R.id.groceries_add)).check(matches(isClickable()));
+    }
+
+    @Test
+    public void removePickedUpAvailable(){
+        onView(withId(R.id.groceries_picked_up_button)).check(matches(isEnabled()));
+        onView(withId(R.id.groceries_picked_up_button)).check(matches(isDisplayed()));
+        onView(withId(R.id.groceries_picked_up_button)).check(matches(isClickable()));
+    }
+
+    @Test
+    public void shopListHasCorrectNumberOfItems() {
+        onView(withId(R.id.groceries_recycler)).check(matches(hasChildCount(1)));
+    }
+
+    @Test
+    public void addingItemShowsNewItem() {
+        addNewItem("item", 2, "kg");
+        // Checking item exists in the view
+        onView(withId(R.id.groceries_recycler)).check(matches(hasChildCount(2)));
+        onView(withId(R.id.groceries_recycler)).check(matches(hasDescendant(withText(containsString("item")))));
+        onView(withId(R.id.groceries_recycler)).check(matches(hasDescendant(withText(containsString("2")))));
+        onView(withId(R.id.groceries_recycler)).check(matches(hasDescendant(withText(containsString("kg")))));
+
+        onView(withId(R.id.groceries_recycler))
+                .perform(RecyclerViewActions.actionOnItemAtPosition(
+                        1,
+                        RecyclerViewHelperActions.clickChildViewWithId(R.id.delete_item_button)));
+    }
+
+    @Test
+    public void navBarTakesBackToMainScreen(){
+        Intents.init();
+        onView(withId(R.id.nav_bar_menu)).perform(click());
+        intended(hasComponent(MainScreenActivity.class.getName()));
+        Intents.release();
+    }
```

Comment:
im not sure how important it is but if you test main screen you can also probably test the calendar screen